### ISSUE TITLE
Reduce on raster broken

### ISSUE THREAD

**Issue by vascotenner** [ioam/holoviews#300]
Sorry for the very short issue:

```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-17-2f3b71189dd3> in <module>()
      1 #zonder.reduce('Scan',np.mean).to.curve(dim_wavelength, 'z').overlay('Power').layout('Location')
      2 ylims = (1, None)
----> 3 zonder[:,:,:300,:].reduce('Scan',np.mean).to.curve(dim_wavelength, 'z', extents=(None, ylims[0], None, ylims[1])).overlay('Power').layout('Location')

/home/tenner/src/holoviews/holoviews/core/spaces.pyc in reduce(self, dimensions, function, **reduce_map)
    277         from ..element import Table
    278         reduced_items = [(k, v.reduce(dimensions, function, **reduce_map))
--> 279                          for k, v in self.items()]
    280         if not isinstance(reduced_items[0][1], Table):
    281             params = dict(util.get_param_values(self.last),

/home/tenner/src/holoviews/holoviews/element/raster.pyc in reduce(self, dimensions, function, **reduce_map)
    148             oidx = self.get_dimension_index(other_dimension[0])
    149             x_vals = self.dimension_values(other_dimension[0].name, unique=True)
--> 150             reduced = reduce_fn(self._zdata, axis=oidx)
    151             data = zip(x_vals, reduced if not oidx else reduced[::-1])
    152             params = dict(dict(self.get_param_values(onlychanged=True)),

NameError: global name 'reduce_fn' is not defined
```

**Comment by philippjfr** [comment]
Thanks for reporting this, must have missed it in the recent refactor. Will get on it tonight.

**Comment by philippjfr** [comment]
Fixed in f57a1d75f3f1d609f1f8781182b4caf8b65f3e85. Sorry about that.
